client/my-sites/domains/domain-search/index.tsx

```diff
@@ -53,62 +53,94 @@ import {
-		productsList: PropTypes.object.isRequired,
-		selectedSite: PropTypes.object,
-		selectedSiteId: PropTypes.number,
-		selectedSiteSlug: PropTypes.string,
-		domainAndPlanUpsellFlow: PropTypes.bool,
-		isDomainAndPlanPackageFlow: PropTypes.bool,
-	};
-
+type DomainSearchProps = {
+	basePath: string;
+	context: Context;
+	domainsWithPlansOnly: boolean;
+	isSiteUpgradeable: boolean | null;
+	productsList: object;
+	selectedSite?: SiteDetails | null;
+	selectedSiteId?: number;
+	selectedSiteSlug: string | null;
+	domainAndPlanUpsellFlow?: boolean;
+	isDomainAndPlanPackageFlow?: boolean;
+	cart: ObjectWithProducts;
+	shoppingCartManager: UseShoppingCart;
+	isAddNewDomainContext: boolean;
+	setCurrentFlowName: ( flowName: string ) => void;
+	recordAddDomainButtonClick: (
+		domainName: string,
+		section: string,
+		position: number,
+		isPremium?: boolean
+	) => void;
+	recordRemoveDomainButtonClick: ( domainName: string ) => void;
+	isSiteOnFreePlan?: boolean;
+	isSiteOnMonthlyPlan: boolean;
+	isDomainUpsell: boolean;
+	currentRoute: string;
+	isFromMyHome: boolean;
+	translate: ReturnType< typeof useTranslate >;
+	isManagingAllDomains: boolean;
+	isEcommerceSite: boolean;
+	preferredView: ReturnType< typeof getSiteOption >;
+	wpAdminUrl: ReturnType< typeof getSiteAdminUrl >;
+};
+
+class DomainSearch extends Component< DomainSearchProps > {
 	isMounted = false;
 
 	state = {
 		domainRegistrationAvailable: true,
 		domainRegistrationMaintenanceEndTime: null,
 	};
 
-	handleDomainsAvailabilityChange = ( isAvailable, maintenanceEndTime = null ) => {
+	handleDomainsAvailabilityChange = (
+		isAvailable: boolean,
+		maintenanceEndTime: number | null = null
+	) => {
 		this.setState( {
 			domainRegistrationAvailable: isAvailable,
 			domainRegistrationMaintenanceEndTime: maintenanceEndTime,
 		} );
 	};
 
-	handleAddRemoveDomain = ( suggestion, position ) => {
+	handleAddRemoveDomain = ( suggestion: DomainSuggestion, position: number ) => {
 		if ( ! hasDomainInCart( this.props.cart, suggestion.domain_name ) ) {
 			this.addDomain( suggestion, position );
 		} else {
 			this.removeDomain( suggestion );
 		}
 	};
 
-	handleAddMapping = ( domain ) => {
-		const domainMappingUrl = domainUseMyDomain( this.props.selectedSiteSlug, {
-			domain,
-			initialMode: useMyDomainInputMode.transferOrConnect,
-		} );
-		this.isMounted && page( domainMappingUrl );
+	handleAddMapping = ( domain: string ) => {
+		if ( this.props.selectedSiteSlug ) {
+			const domainMappingUrl = domainUseMyDomain( this.props.selectedSiteSlug, {
+				domain,
+				initialMode: useMyDomainInputMode.transferOrConnect,
+			} );
+			this.isMounted && page( domainMappingUrl );
+		}
 	};
 
-	handleAddTransfer = async ( domain ) => {
+	handleAddTransfer = async ( domain: string ) => {
 		try {
-			await this.props.shoppingCartManager.addProductsToCart( [ domainTransfer( { domain } ) ] );
+			await this.props.shoppingCartManager.addProductsToCart( [
+				domainTransfer( { domain, extra: {} } ),
```

Comment:
This empty object works, but I don't really like it.

`domainTransfer` requires an `extra` prop, but [all of the properties of that interface are optional](https://github.com/Automattic/wp-calypso/blob/c2c6ec18d29fa98a36b771c68b2bba92a4ee7b00/packages/shopping-cart/src/types.ts#L649-L680). I'm tempted to make the `extra` param of this function optional itself, rather than being forced to pass this empty object. Thoughts?

Reply:
I belive that we can safely update `domainTransfer` function, by setting `extra` as optional value. It should not make any issues in the code. And additionally the function expects extra as optional, since inside we have ternary operator `...( properties.extra ? { extra: properties.extra } : {} ),`, so let's proceed with updating `domainTransfer` function and remove this empty object `{}`.